app/models/speedrun_dot_com_user.rb

```diff
@@ -0,0 +1,26 @@
+require 'speedrundotcom'
+
+class SpeedrunDotComUser < ApplicationRecord
+  belongs_to :user
+
+  def self.from_api_key!(api_key, user:)
+    srdc_user = SpeedrunDotComUser.find_by(api_key: api_key)
+    return srdc_user if srdc_user.present?
+
+    result = SpeedrunDotCom::User.from_api_key(api_key)
+    return if result.nil?
+
+    new(srdc_id: result['id'], api_key: api_key, user: user).from_result!(result)
+  end
+
+  def sync!
+    from_result!(SpeedrunDotCom::User.from_id(srdc_id))
+  end
+
+  def from_result!(result)
+    update(
+      name:           result['names']['international'],
+      url:            result['weblink'],
+    ) && self
```

Comment:
Did we ever figure out if the instance version of `update(...)` will return a true/false value or the instance itself?

Reply:
Yeah, I tested it and it returns `true`/`false`, so this `&& self` is still necessary.

Reply:
But if it returns false, wouldn't that immediately return the false value instead of self like you want?

Reply:
Yeah, that's intended. We want to return something falsey if the operation failed, or the object itself if it succeeded.

Reply:
Oh, that's my bad then.  I thought the intention was to return the object itself no matter what.